components/rabbitmq/scripts/start.py

```diff
@@ -12,11 +12,22 @@
 import utils  # NOQA
 
 RABBITMQ_SERVICE_NAME = 'rabbitmq'
+ctx_properties = utils.ctx_factory.get(RABBITMQ_SERVICE_NAME)
+rabbitmq_endpoint_ip = ctx_properties['rabbitmq_endpoint_ip']
+PORT = 5671 if ctx_properties['rabbitmq_ssl_enabled'] else 5672
 
 
-ctx_properties = utils.ctx_factory.get(RABBITMQ_SERVICE_NAME)
+@utils.retry(ValueError)
+def check_rabbit_running():
+    """Use rabbitmqctl status to check if RabbitMQ is working.
 
-rabbitmq_endpoint_ip = ctx_properties['rabbitmq_endpoint_ip']
+    Sometimes rabbit takes a while to start, so this is retried several times.
+    Note that this is currently impossible to do on a remote host, so
+    this check only runs when rabbitmq is installed locally.
+    """
+    result = utils.sudo(['rabbitmqctl', 'status'], ignore_failures=True)
```

Comment:
in the future we might want to look into getting `rabbitmqctl status` working also remotely, which will allow us to use this check also for external rabbitmq, instead of the - less strict - `verify_port_open` check